pavelib/quality.py

```diff
@@ -258,6 +264,16 @@ def run_quality(options):
     pep8_reports = u' '.join(pep8_files)
 
     try:
+        violations_str = None
+        with open(pep8_reports, 'r') as f:
+            violations_list = f.readlines()
+            violations_str = '\n'.join(violations_list)
+
+        if violations_str:
+            raise BuildFailure(
+                'Number of pep8 violations should be 0; found these violations:\n' + violations_str
+            )
+
         sh(
             "diff-quality --violations=pep8 {pep8_reports} {percentage_string} "
```

Comment:
I'm wondering if, rather than calling diff-quality here, we start by just writing the pep8 output to `{dquality_dir}/diff_quality_pep8.html`.  That will get it showing up in the Diff Quality report accurately. Then later figure out how to make it as readable as the html generated by diff-quality.